framework/src/parser/Parser.C

```diff
@@ -1206,6 +1206,13 @@ Parser::extractParams(const std::string & prefix, InputParameters & p)
     if (node && node->type() == hit::NodeType::Field)
     {
       p.set_attributes(it.first, false);
+
+      // Check if we have already printed the deprec param message.
+      // If we haven't, add it to the tracker, and print it.
+      if (!_deprec_param_tracker.count(it.first))
+        if (p.attemptPrintDeprecated(it.first))
```

Comment:
Do we really need to know if it was printed or not?  If it wasn't... then it won't be in the future... right?

Reply:
I'm confused what you mean here. All that happens here is that it checks if the message has been printed for that variable.